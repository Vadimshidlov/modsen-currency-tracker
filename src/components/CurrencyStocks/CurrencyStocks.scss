@use "@/constants/styles/fonts" as fonts;
@use "@/constants/styles/colors" as colors;
@import "@/index.scss";

.stocks__container {
    margin: 0 auto;
    margin-bottom: 86px;
    max-width: 1224px;
    .stocks__title {
        margin-bottom: 24px;

        @include fonts.PoppinsLight;
        color: colors.$color_white;
        font-size: fonts.$font-size_32;
        line-height: 41.14px;
        // border-bottom: 2px solid colors.$currency_card_border;
        // border-bottom-width: 520px;
    }

    .title__border {
        margin-bottom: 50px;
        background-color: colors.$currency_card_border;
        height: 2px;
        width: 520px;
    }

    .stocks__items {
        display: flex;
        column-gap: 184px;
        flex-wrap: wrap;
    }
}
