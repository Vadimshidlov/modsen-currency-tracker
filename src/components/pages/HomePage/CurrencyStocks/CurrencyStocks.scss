@use "@/constants/styles/fonts" as fonts;
@use "@/constants/styles/colors" as colors;
//@import "@/index.scss";

.stocks__container {
  max-width: 1224px;
  //margin: 0 auto 86px;
  margin-bottom: 86px;

  @media (max-width: 500px) {
    margin-bottom: 32px;
  }

  .stocks__title {
    margin-bottom: 24px;

    @include fonts.PoppinsLight;
    color: colors.$color_white;
    font-size: fonts.$font-size_32;
    line-height: 41.14px;

    @media (max-width: 768px) {
      margin-bottom: 12px;
      font-size: fonts.$font-size_24;
      line-height: 22px;
    }

    @media (max-width: 500px) {
      margin-bottom: 12px;
      font-size: fonts.$font-size_18;
      line-height: 17px;
    }

    @media (max-width: 450px) {
      margin-bottom: 7px;
      font-size: fonts.$font-size_12;
      line-height: 15px;
    }
  }

  .title__border {
    margin-bottom: 50px;
    background-color: colors.$currency_card_border;
    height: 2px;
    width: 520px;

    @media (max-width: 1120px) {
      margin-bottom: 35px;
      width: 100%;
    }

    @media (max-width: 500px) {
      margin-bottom: 16px;
      width: 100%;
    }
  }

  .stocks__items {
    display: flex;
    //column-gap: 184px;
    justify-content: space-between;
    row-gap: 56px;
    flex-wrap: wrap;

    @media (max-width: 1120px) {
      display: flex;
      flex-direction: column;
      row-gap: 30px;
    }

    @media (max-width: 768px) {
      display: flex;
      flex-direction: column;
      row-gap: 20px;
    }
  }

  &.light {
    .stocks__title {
      color: colors.$black_background;
    }

    .currency-card__container {
      background: colors.$currency-card_background__light;
    }

    .currency-card__container {
      .currency-card__titile {
        color: colors.$black_background;
      }

      .currency-card__percent {
        color: colors.$black_background;
      }
    }
  }
}
