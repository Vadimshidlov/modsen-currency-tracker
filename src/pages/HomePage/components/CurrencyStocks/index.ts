export { default } from "@/pages/HomePage/components/CurrencyStocks/CurrencyStocks";
