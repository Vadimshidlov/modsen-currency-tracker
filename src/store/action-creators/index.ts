export { getMapData } from "@/store/action-creators/getMapData";
export { getCurrency } from "@/store/action-creators/getCurrency";
export {
    selectSecondCurrency,
    closeModalWindow,
    openModalWindow,
} from "@/store/action-creators/openModalWindow";
export { switchLightTheme, switchDarkTheme } from "@/store/action-creators/toggleTheme";
export {
    errorAuthUserLocation,
    successAuthUserLocation,
} from "@/store/action-creators/authUserLocation";
