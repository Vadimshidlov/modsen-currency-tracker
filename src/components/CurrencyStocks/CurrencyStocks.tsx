import React from "react";
import Text from "@/components/Text/Text";
import CurrencyCard from "@/components/CurrencyCard/CurrencyCard";
import "@/components/CurrencyStocks/CurrencyStocks.scss";

export default function CurrencyStocks() {
    return (
        <div className="stocks__container">
            <Text className="stocks__title title">Stocks</Text>
            <div className="title__border" />
            <div className="stocks__items">
                {[0, 1].map(() => (
                    <CurrencyCard />
                ))}
            </div>
        </div>
    );
}
