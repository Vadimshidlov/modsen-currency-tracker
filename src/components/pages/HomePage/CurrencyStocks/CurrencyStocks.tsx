import React from "react";
import Text from "@/components/pages/HomePage/Text/Text";
import "@/components/pages/HomePage/CurrencyStocks/CurrencyStocks.scss";
import ModalWindow from "@/components/pages/HomePage/ModalWindow/ModalWindow";
import { mockStockData } from "@/store/currency-data/mockStockData";
import StockCard from "@/components/pages/HomePage/StockCard/StockCard";

export default function CurrencyStocks() {
    return (
        <div className="stocks__container">
            <Text className="stocks__title title">Stocks</Text>
            <div className="title__border" />
            <div className="stocks__items">
                {mockStockData.map((stockItem) => (
                    <StockCard
                        stockCode={stockItem.stockCode}
                        value={stockItem.stockValue}
                        key={stockItem.stockCode}
                    />
                ))}
            </div>
            <ModalWindow isOpen={false} />
        </div>
    );
}
