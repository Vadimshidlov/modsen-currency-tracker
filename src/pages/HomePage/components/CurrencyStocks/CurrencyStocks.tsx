import React from "react";
import Text from "@/pages/HomePage/components/Text/index";
import "@/pages/HomePage/components/CurrencyStocks/CurrencyStocks.scss";
import ModalWindow from "@/pages/HomePage/components/ModalWindow/index";
import { mockStockData } from "@/store/currency-data/index";
import StockCard from "@/pages/HomePage/components/StockCard/index";
import { useTypedSelectorHook } from "@/store/hooks/index";
import { selectTheme } from "@/store/selectors/index";

export default function CurrencyStocks() {
    const { theme } = useTypedSelectorHook(selectTheme);

    return (
        <section className={theme === "Light" ? "stocks__container light" : "stocks__container"}>
            <Text className="stocks__title title">Stocks</Text>
            <div className="title__border" />
            <div className="stocks__items">
                {mockStockData.map((stockItem) => (
                    <StockCard
                        stockCode={stockItem.stockCode}
                        value={stockItem.stockValue}
                        key={stockItem.stockCode}
                    />
                ))}
            </div>
            <ModalWindow isOpen={false} />
        </section>
    );
}
