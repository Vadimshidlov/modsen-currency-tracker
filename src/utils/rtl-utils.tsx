import React, { PropsWithChildren } from "react";
import { Store, applyMiddleware, createStore } from "redux";
import { thunk } from "redux-thunk";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { Provider } from "react-redux";
import { RootStateType, rootReducer } from "@/store/reducers";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
    preloadedState?: Partial<RootStateType>;
    store?: Store;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        store = createStore(rootReducer, preloadedState, applyMiddleware(thunk)),
        ...renderOptions
    }: ExtendedRenderOptions = {},
) {
    function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
        return <Provider store={store}>{children}</Provider>;
    }
    return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from "@testing-library/react";

export { renderWithProviders as render };
