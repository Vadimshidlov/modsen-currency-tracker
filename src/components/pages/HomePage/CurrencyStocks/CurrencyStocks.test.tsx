import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "@/store";
import { renderWithProviders, screen } from "@/utils/rtl-utils";
import CurrencyStocks from "@/components/pages/HomePage/CurrencyStocks/CurrencyStocks";

describe("CurrencyStocks component tests", () => {
    it("should correct render component", async () => {
        renderWithProviders(<CurrencyStocks />);

        expect(screen.getByText("Stocks")).toBeInTheDocument();
        expect(screen.getByText("Bovespa Index")).toBeInTheDocument();
        expect(screen.queryAllByText("0.15%")).toHaveLength(2);
        expect(screen.getByText("IFIX")).toBeInTheDocument();
    });

    it("snapshot matches", async () => {
        const { asFragment } = renderWithProviders(<CurrencyStocks />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should match snapshot", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <CurrencyStocks />
                </Provider>,
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
