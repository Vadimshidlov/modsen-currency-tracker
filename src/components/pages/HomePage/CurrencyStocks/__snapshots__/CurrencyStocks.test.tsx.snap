// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyStocks component tests should match snapshot 1`] = `
<div
  className="stocks__container"
>
  <p
    className="stocks__title title"
  >
    Stocks
  </p>
  <div
    className="title__border"
  />
  <div
    className="stocks__items"
  >
    <button
      className="currency-card__container"
      disabled={true}
    >
      <svg
        className="currency-card__icon"
      />
      <div
        className="currency-card__info"
      >
        <p
          className="currency-card__titile"
        >
          Bovespa Index
        </p>
        <p
          className="currency-card__percent"
        >
          0.15%
        </p>
      </div>
    </button>
    <button
      className="currency-card__container"
      disabled={true}
    >
      <svg
        className="currency-card__icon"
      />
      <div
        className="currency-card__info"
      >
        <p
          className="currency-card__titile"
        >
          IFIX
        </p>
        <p
          className="currency-card__percent"
        >
          0.15%
        </p>
      </div>
    </button>
  </div>
</div>
`;

exports[`CurrencyStocks component tests snapshot matches 1`] = `
<DocumentFragment>
  <div
    class="stocks__container"
  >
    <p
      class="stocks__title title"
    >
      Stocks
    </p>
    <div
      class="title__border"
    />
    <div
      class="stocks__items"
    >
      <button
        class="currency-card__container"
        disabled=""
      >
        <svg
          class="currency-card__icon"
        />
        <div
          class="currency-card__info"
        >
          <p
            class="currency-card__titile"
          >
            Bovespa Index
          </p>
          <p
            class="currency-card__percent"
          >
            0.15%
          </p>
        </div>
      </button>
      <button
        class="currency-card__container"
        disabled=""
      >
        <svg
          class="currency-card__icon"
        />
        <div
          class="currency-card__info"
        >
          <p
            class="currency-card__titile"
          >
            IFIX
          </p>
          <p
            class="currency-card__percent"
          >
            0.15%
          </p>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
